### DISCUSSION TITLE
Make it easy to get the correct file path of a dataset

### DISCUSSION THREAD

**Discussion by merelcht** [kedro-org/kedro#3753] [Idea]
### Description
To get the correct (versioned) file path for a dataset is quite hard. There doesn't seem to be one way that works for both non-versioned and versioned datasets and/or local and remote datasets.

The main question here is _why_ users need to access the file path. Not all datasets have a file path, e.g. `APIDataset` and so it's  important to understand the true user need, before diving into solution.

### Context

1. `dataset._filepath()` for non-versioned, local dataset
2. `dataset_get_load_path()` for versioned, local datasets
3. Remote datasets: `get_filepath_str(self._get_load_path(), self._protocol)` 

This might not even be the full list of ways to get the file path.

> [!IMPORTANT]
> This idea is based on observations from several Kedro engineers see e.g. https://github.com/kedro-org/kedro/issues/1778. However, we need a clear view on what user needs are when it comes to why they need the file path and what their use cases are. Any implementation should be preceded by user research: https://github.com/kedro-org/kedro/issues/1978

**Comment by astrojuanlu** [comment]
@Galileo-Galilei mentioned this idea in the context of a user problem with MLflow https://kedro.hall.community/kedro-mlflow-netcdf-dataset-path-issue-dLxZxJzh1dsl
